tests/test_parser.py

```diff
@@ -1517,3 +1517,13 @@ def test_alt_dockerfile_names(self, tmpdir):
 
         validate = DockerfileParser(path=tmpdir, dockerfile_filename="Containerfile")
         assert validate.baseimage == out.baseimage
+
+    def test_dockerfile_path_compatibility(self, tmpdir):
+        parser = DockerfileParser(path=tmpdir)
+        assert str(parser.dockerfile) == parser.dockerfile_path
+        assert parser.dockerfile == tmpdir / "Dockerfile"
```

Comment:
nitpick: `tmpdir` is not a pathlib.Path, it is a py.path.local https://docs.pytest.org/en/7.4.x/how-to/tmp_path.html#the-tmpdir-and-tmpdir-factory-fixtures

It looks like Paths and py.paths are considered equal though, so the test still works